main.py

```diff
@@ -15,22 +15,24 @@ def signedOn(self):
     def privmsg(self, user, channel, msg):
         nick = self.nickname
         user = user.split('!')[0]
-        if msg == '~pym':
-            self.msg(channel, 'http://pymbook.readthedocs.org/en/latest/')
-        if msg == '{}, hello'.format(nick) or msg == '{}: hello'.format(nick):
-            self.msg(channel, '{}, hello!'.format(user))
-        if msg == '{}, ping'.format(nick) or msg == '{}: ping'.format(nick):
-            self.msg(channel, '{}, pong'.format(user))
-        if msg == '{}, date'.format(nick) or msg == '{}: date'.format(nick) \
-                or msg == '~date':
-            self.msg(channel, "Current date & time " + time.strftime("%c"))
-        if msg == '~random':
-            clf = urllib2.urlopen(
-                "http://www.commandlinefu.com/commands/random/plaintext").read()
-            self.msg(channel, clf.split("\n\n")[1])
-        if msg.startswith("~whatis"):
-            self.msg(channel, wikipedia.summary(
-                msg[msg.index(" ") + 1:], sentences=1).encode('utf-8'))
+        with open(sys.argv[1],'a') as fobj:
+            fobj.write('\n' + time.strftime('[%d-%m-%Y %H:%M:%S]') + ' ' + '<'+ user + '>' + ' ' + msg)
+            if msg == '~pym':
```

Comment:
You don't need to put everything under `with`. This will only slow down the bot's response, indent out every `if` statement.